### PR TITLE
unit test for create and edit vendor

### PR DESCRIPTION
**Pull request by RevathyRajeevan**
Added unit testing for vendor creation and editing
image

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -7,4 +8,5 @@ import { Component } from '@angular/core';
 })
 export class AppComponent {
   title = 'ILP_Backroads';
-}
+  constructor(private router: Router) {}
```
Comment (athul-sanjeevan) [inline src/app/app.component.ts]: do we need this?

File: src/app/features/vendor/vendor-creation/vendor-creation.component.html [hunk 2]
```diff
@@ -1,202 +1,214 @@
...
+      id="submit-btn"
+      label="Submit"
+      [disabled]="addVendorForm.invalid"
+      class="submit-btn"
+    />
+    <button
+      id="cancel-btn"
+      type="button"
+      (click)="
+        confirmDialog.showConfirmDialog(
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.html]: make it as a method

File: src/app/features/vendor/vendor-creation/vendor-creation.component.spec.ts [hunk 3]
```diff
@@ -1,21 +1,231 @@
...
+    component.isEdit = false;
+    component.addVendorForm = getVendorServiceFormMock();
+    component.submitVendor();
+    expect(spyObj.addVendorSpy).toHaveBeenCalled();
+  });
+
+  it('should display error when they are present during vendor form submission.', () => {
+    const { component, spyObj } = setup();
+    component.addVendorForm = getVendorServiceFormMock();
+    const errorResponse = {
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.spec.ts]: make this as method and reuse this error response

File: src/app/shared/confirm-dialog/confirm-dialog.component.css [hunk 4]
```diff
@@ -0,0 +1,13 @@
+  ::ng-deep .p-confirm-dialog-reject {
+    background-color: white;
+    color: red;
+  }
+   
+  ::ng-deep .p-confirm-dialog .p-dialog-content i {
+    color: orange;
+  }
+  ::ng-deep .p-dialog-footer .p-element {
```
Comment (athul-sanjeevan) [inline src/app/shared/confirm-dialog/confirm-dialog.component.css:10]: line gap before this

File: src/app/features/vendor/vendor-creation/vendor-creation.component.spec.ts [hunk 5]
```diff
@@ -1,21 +1,231 @@
...
+  it('should update the vendor data when update form is submitted', () => {
+    const { component, spyObj } = setup();
+    component.isEdit = true;
+    component.selectedVendorId = 1;
+    component.addVendorForm = getVendorServiceFormMock();
+    component.submitVendor();
+    expect(spyObj.updateVendorSpy).toHaveBeenCalled();
+  });
+
+  it('should show error during updating the vendor data', () => {
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.spec.ts]: when?

File: src/app/app.component.ts (hunk 1 above)
Comment (RevathyRajeevan) [reply src/app/app.component.ts]: removed

File: src/app/features/vendor/vendor-creation/vendor-creation.component.html (hunk 2 above)
Comment (RevathyRajeevan) [reply src/app/features/vendor/vendor-creation/vendor-creation.component.html]: done

File: src/app/features/vendor/vendor-creation/vendor-creation.component.spec.ts (hunk 3 above)
Comment (RevathyRajeevan) [reply src/app/features/vendor/vendor-creation/vendor-creation.component.spec.ts]: done

File: src/app/shared/confirm-dialog/confirm-dialog.component.css (hunk 4 above)
Comment (RevathyRajeevan) [reply src/app/shared/confirm-dialog/confirm-dialog.component.css:10]: done

File: src/app/features/vendor/vendor-creation/vendor-creation.component.spec.ts (hunk 5 above)
Comment (RevathyRajeevan) [reply src/app/features/vendor/vendor-creation/vendor-creation.component.spec.ts]: changed

File: src/app/app.component.ts [hunk 6]
```diff
@@ -1,4 +1,5 @@
 import { Component } from '@angular/core';
+import { NavigationStart, Router } from '@angular/router';
```
Comment (athul-sanjeevan) [inline src/app/app.component.ts]: remove the imports if not need

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts [hunk 7]
```diff
@@ -63,7 +65,10 @@ export class VendorCreationComponent implements OnInit {
   markets!: IDropDownFields[];
   services!: IDropDownFields[];
   selectedVendorId!: number;
+  confirmationMessage='';
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: format

File: src/app/shared/confirm-dialog/confirm-dialog.component.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    message: this.message,
+    accept: () => {
+        this.onConfirm.emit();
+      }
+    });
+  }
+  }
+
+
+
```
Comment (athul-sanjeevan) [inline src/app/shared/confirm-dialog/confirm-dialog.component.ts]: remove unwanted lines

File: src/app/shared/confirm-dialog/confirm-dialog.component.ts [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+  ngOnInit()
+  {   
+    this.confirmationService.confirm({
+    message: this.message,
+    accept: () => {
+        this.onConfirm.emit();
+      }
+    });
+  }
+  }
```
Comment (athul-sanjeevan) [inline src/app/shared/confirm-dialog/confirm-dialog.component.ts]: correct the bracket nesting

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts [hunk 10]
```diff
@@ -63,7 +60,9 @@ export class VendorCreationComponent implements OnInit {
   markets!: IDropDownFields[];
   services!: IDropDownFields[];
   selectedVendorId!: number;
-  
+  confirmationMessage='';
+  isViewConfirmModal=false
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: format & rename the variable as isConfirmPopupVisible or similar

**Review by athul-sanjeevan — CHANGES_REQUESTED** [review]

File: src/app/app.component.ts (hunk 6 above)
Comment (RevathyRajeevan) [reply src/app/app.component.ts]: done

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts (hunk 7 above)
Comment (RevathyRajeevan) [reply src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: done

File: src/app/shared/confirm-dialog/confirm-dialog.component.ts (hunk 8 above)
Comment (RevathyRajeevan) [reply src/app/shared/confirm-dialog/confirm-dialog.component.ts]: done

File: src/app/shared/confirm-dialog/confirm-dialog.component.ts (hunk 9 above)
Comment (RevathyRajeevan) [reply src/app/shared/confirm-dialog/confirm-dialog.component.ts]: done

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts (hunk 10 above)
Comment (RevathyRajeevan) [reply src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: done

**Review by athul-sanjeevan — APPROVED** [review]

**Pull request merged**